iothub/device/src/Serialization/DefaultPayloadConvention.cs

```diff
@@ -22,9 +44,48 @@ private DefaultPayloadConvention()
         public static DefaultPayloadConvention Instance { get; } = new DefaultPayloadConvention();
 
         /// <inheritdoc/>
-        public override PayloadSerializer PayloadSerializer { get; } = NewtonsoftJsonPayloadSerializer.Instance;
+        public override string ContentType => "application/json";
 
         /// <inheritdoc/>
-        public override PayloadEncoder PayloadEncoder { get; } = Utf8PayloadEncoder.Instance;
+        public override string ContentEncoding => s_encoding.WebName;
+
+        /// <inheritdoc/>
+        public override byte[] GetObjectBytes(object objectToSendWithConvention)
+        {
+            string payloadString = Serialize(objectToSendWithConvention);
+            return s_encoding.GetBytes(payloadString);
+        }
+
+        /// <inheritdoc/>
+        public override T GetObject<T>(byte[] objectToConvert)
+        {
+            if (objectToConvert is null)
+            {
+                return default;
+            }
+
+            string payloadString = s_encoding.GetString(objectToConvert);
+            return GetObject<T>(payloadString);
+        }
+
+        /// <inheritdoc/>
+        public override Task<T> GetObjectAsync<T>(Stream streamToConvert)
+        {
+            using var sw = new StreamReader(streamToConvert);
+            using var jtr = new JsonTextReader(sw);
+            return Task.FromResult(s_jsonSerializer.Deserialize<T>(jtr));
```

Comment:
Is there an implementation where we expect this to be async? I'm wondering why we have this declared as async if we have to do a Task.FromResult anyway?

Reply:
STJ requires this be async so I changed PayloadConvention interface to be Task<T>.

I could revert that and simply use `GetAwaiter().GetResult()` here. At the time, I tried to make it async all the way up the stack, until I found the AMQP layer isn't async where I need it to be.

However, it eventually will be. We need the new major version to come out.

Thoughts?